.github/workflows/msbuild.yml

```diff
@@ -43,7 +43,7 @@ jobs:
     - name: Nuget Restore Dafny
       run: nuget restore dafny/Source/Dafny.sln
       env:
-        MONO_TLS_PROVIDER: legacy
+        MONO_TLS_PROVIDER: thisshouldfail
```

Comment:
Correct error message is shown per https://github.com/mono/mono/pull/17391/files
This implies the ENV_VAR is getting passed in as expected. If we used legacy, this would just be treated as default per linked PR.